catalog/validator/catalog-validator-unsupportedattributes/src/test/java/org/codice/ddf/validator/query/unsupportedattributes/UnsupportedAttributeQueryValidatorTest.java

```diff
@@ -0,0 +1,231 @@
+
+    assertThat(violations, hasSize(1));
+    for (QueryValidationViolation violation : violations) {
+      assertThat(violation.getSeverity(), is(Severity.ERROR));
+      assertThat(
+          violation.getMessage(),
+          is("The field \"attr-2\" is not supported by the src-1 Content Store(s)"));
+      Map<String, Object> extraData = violation.getExtraData();
+      assertThat(extraData.get("attribute"), is("attr-2"));
+      assertThat((Set<String>) extraData.get("sources"), Matchers.hasItems("src-1"));
+    }
+  }
+
+  @Test
+  public void testValidateNoViolationsIfSourceDoesNotHaveRestriction()
+      throws UnsupportedQueryException {
+    List<SourceAttributeRestriction> restrictions =
+        ImmutableList.of(mockRestriction("src-2", ImmutableSet.of("attr-2")));
+    Mockito.when(attributeExtractor.extractAttributes(any())).thenReturn(ImmutableSet.of("attr-2"));
+    QueryRequest request = mockQueryRequest("src-1");
+    validator = new UnsupportedAttributeQueryValidator(attributeExtractor);
+    validator.setSourceAttributeRestrictions(restrictions);
+
+    Set<QueryValidationViolation> violations = validator.validate(request);
+
+    assertThat(violations, hasSize(0));
+  }
+
+  @Test
+  public void testValidateNoViolationsIfFailedToGetAttributesFromQuery()
+      throws UnsupportedQueryException {
+    List<SourceAttributeRestriction> restrictions =
+        ImmutableList.of(mockRestriction("src-1", ImmutableSet.of("attr-1")));
+    Mockito.when(attributeExtractor.extractAttributes(any()))
+        .thenThrow(UnsupportedQueryException.class);
+    QueryRequest request = mockQueryRequest("src-1");
+    validator = new UnsupportedAttributeQueryValidator(attributeExtractor);
+    validator.setSourceAttributeRestrictions(restrictions);
+
+    Set<QueryValidationViolation> violations = validator.validate(request);
+
+    assertThat(violations, hasSize(0));
+  }
+
+  @Test
+  public void testValidateMultipleInvalidAttributes() throws UnsupportedQueryException {
+    List<SourceAttributeRestriction> restrictions =
+        ImmutableList.of(mockRestriction("src-1", ImmutableSet.of("attr-1")));
+    Mockito.when(attributeExtractor.extractAttributes(any()))
+        .thenReturn(ImmutableSet.of("attr-2", "attr-3"));
+    QueryRequest request = mockQueryRequest("src-1");
+    validator = new UnsupportedAttributeQueryValidator(attributeExtractor);
+    validator.setSourceAttributeRestrictions(restrictions);
+
+    Set<QueryValidationViolation> violations = validator.validate(request);
+
+    assertThat(violations, hasSize(2));
+    for (QueryValidationViolation violation : violations) {
+      assertThat(violation.getSeverity(), is(Severity.ERROR));
+      Map<String, Object> extraData = violation.getExtraData();
+      assertThat((Set<String>) extraData.get("sources"), Matchers.hasItems("src-1"));
+      String attribute = (String) extraData.get("attribute");
+      if (attribute.equals("attr-2")) {
```

Comment:
✏️ We should avoid conditional logic in unit tests where possible.
```java
assertThat(violation.getMessage(), is("The field \"" + attribute + "\" is not supported by the src-1 Content Store(s)"));
```

Reply:
Swapped the conditional logic for a `hasItem` call instead